tests/integration/netdev.bats

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	requires root
+	setup_busybox
+	# create a dummy interface to move to the container
+	ip link add dummy0 type dummy
+}
+
+function teardown() {
+	ip link del dev dummy0
+	teardown_bundle
+}
+
+@test "move network device to container network namespace" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+								| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+}
+
+@test "move network device to container network namespace and restore it back" {
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	update_config ' .linux.netDevices |= {"dummy0": {} }'
+
+	runc run -d --console-socket "$CONSOLE_SOCKET" test_busybox
+	[ "$status" -eq 0 ]
+
+	# the network namespace owner controls the lifecycle of the interface
+	ip netns exec "$ns_name" ip link set dev dummy0 netns 1
+	[ "$status" -eq 0 ]
+
+	runc delete --force test_busybox
+
+	# verify the interface is back in the root network namespace
+	ip address show dev dummy0
+	[ "$status" -eq 0 ]
+}
+
+@test "move network device to precreated container network namespace" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+								| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+
+	# verify the interface is still present in the network namespace
+	ip netns exec "$ns_name" ip address show dev dummy0
+	[ "$status" -eq 0 ]
+
+	ip netns del "$ns_name"
+}
+
+@test "move network device to precreated container network namespace and set ip address" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+								| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	# set a custom address to the interface
+	# set the interface down to avoid network problems
+	ip link set down dev dummy0
+	ip address add 169.254.169.77 dev dummy0
+
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
```

Comment:
Wanna move this to a helper function in this file, maybe? It seems a lot of tests are doing the same

Reply:
See also https://github.com/opencontainers/runc/pull/4538/commits/6b0bf8b30cca7e3ff6b1e016d374df96f5879bbd#r2056759847

Reply:
Not sure what you mean, github doesn't show any discussions with that link and the commit doesn't have any comments when I open it here. What am I missing?

Reply:
added the network namespace creation and deletion logic to the `setup` and `teardown` hooks